workload/cpu_fold.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Samir A Mulani <samir@linux.vnet.ibm.com>
+
+from avocado import Test
+from avocado.utils import process
+from avocado.utils.software_manager.manager import SoftwareManager
+import os
+
+
+class cpu_workload(Test):
+    """
+    This test is basically run the cpu workload as a daemon process,
+    -> Covers all different cpu off on scenarios for single cpu,
+    multiple cpu and multiple cycles of cpu toggle.
+    """
+
+    def setUp(self):
+        """
+        Check required packages is installed.
+        """
+        sm = SoftwareManager()
+        for pkg in ['util-linux', 'powerpc-utils', 'numactl']:
+            if not sm.check_installed(pkg) and not sm.install(pkg):
+                self.cancel("%s is required to continue..." % pkg)
+
+    def test_cpu_start(self):
+        """
+        Start the CPU Workload
+        """
+        self.cwd = os.getcwd()
+        self.smt_workload = self.cwd + "/" + \
```

Comment:
please have variable scope as local , even rename it logical name as this for cpu_folding